src/ecu.c

```diff
@@ -78,6 +82,92 @@ void default_state()
   status.pwm1_out = config.pwm1_min;
 }
 
+/*
+  set variable
+ */
+static void command_set(char *arg1, const char *arg2)
+{
+    if (!arg2) {
+        // with no 2nd arg show var
+        config_show(arg1);
+    } else {
+        config_set(arg1, arg2);
+    }
+}
+
+/*
+  change display period
+ */
+static void command_period(const char *arg)
+{
+    uint16_t new_period = strtoul(arg, NULL, 10);
+    if (new_period < 50 || new_period > 5000) {
+        logmsgf("Invalid period %u", new_period);
+    } else {
+        telem_period_ms = new_period;
+        logmsgf("new period %u", new_period);
+    }
+}
+
+/*
+  process one line of input
+ */
+static void process_line(char *line)
+{
+    char *cmd = strtok(line, " ");
+    if (!cmd) {
+        return;
+    }
+    if (strcmp(cmd, "config") == 0) {
+        char *arg = strtok(NULL, " ");
+        if (arg) {
+            if (strcmp(arg, "defaults") == 0) {
+                config_defaults();
+                logmsgf("config reset to defaults");
+            }
+        } else {
+            // show config
+            config_dump();
+        }
+    } else if (strcmp(cmd, "period") == 0) {
+        char *arg = strtok(NULL, " ");
+        if (arg) {
+            command_period(arg);
+        }
+    } else if (strcmp(cmd, "set") == 0) {
+        char *arg1 = strtok(NULL, " ");
+        char *arg2 = strtok(NULL, " ");
+        if (arg1) {
+            command_set(arg1, arg2);
+        }
+    }
+}

```

Comment:
Probably need a "config save" command to commit to eeprom.

config_load() has been commented out in the code a few lines below. We should restore this..